### PR TITLE
feat(cb2-4089): amend test type

### PR DESCRIPTION
**Pull request by gjulien-bjss**
## Amend test type

_This story covers test type amendments between PSV test type IDs for same test type validation group 1._
[CB2-4089](https://dvsa.atlassian.net/browse/CB2-4089)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,6 +79,7 @@
     "@storybook/testing-library": "^0.0.9",
     "@types/jest": "^27.4.1",
     "@types/json-server": "^0.14.4",
+    "@types/lodash.clonedeep": "^4.5.7",
```
Comment (mcadenas-bjss) [inline package.json:82]: Why is this dep in twice? maybe remove this one.

Comment (gjulien-bjss) [reply package.json:82]: hum... it's types

File: src/app/forms/components/test-type-select/test-type-select.component.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    </ol>
+  </div>
+  <div
+    *ngFor="let category of categories; trackBy: tackByFn; let i = index"
+    class="list_container"
+    [ngClass]="{ 'list_container--shrink': categories.length > i + 2 }"
+  >
+    <h2 class="govuk-heading-m">Browse: {{ category.name }}</h2>
+    <ol class="list">
+      <!-- <ol *ngFor="let category of categories; trackBy: tackByFn; let i = index" class="list"> -->
```
Comment (kpearmanbjss) [inline src/app/forms/components/test-type-select/test-type-select.component.html:40]: Is this commented out line needed for something or should it have been removed?

**Review by kpearmanbjss — APPROVED** [review]

**Review by Ruairidh-BJSS — APPROVED** [review]

File: src/app/forms/components/test-type-select/test-type-select.component.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+<div *ngIf="edit" class="categories">
+  <div class="list_container" [ngClass]="{ 'list_container--shrink': categories.length > 0 + 1 }">
+    <ol class="list">
+      <li
+        *ngFor="let type of selectAllTestTypes$ | async; trackBy: tackByFn"
+        class="list__item"
+        [ngClass]="{ 'list__item--active': isSelected(type.id!!) }"
+      >
+        <button (click)="handleCategory(type, 0)">
+          {{ type.id }}: {{ type.name }} <span [ngClass]="{ 'list__item--more': hasNext(type) }"></span>
```
Comment (andrewf-bjss) [inline src/app/forms/components/test-type-select/test-type-select.component.html:28]: Do we need to show the TestTypeId (`type.id`) to the user?  (and again on line 47)

```suggestion
          {{ type.name }} <span [ngClass]="{ 'list__item--more': hasNext(type) }"></span>
```

**Review by andrewf-bjss — APPROVED** [review]

File: src/app/forms/components/test-type-select/test-type-select.component.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <h2 class="govuk-heading-m">Browse: {{ category.name }}</h2>
+    <ol class="list">
+      <ng-container *ngIf="hasNext(category)">
+        <li
+          *ngFor="let type of category.nextTestTypesOrCategories; trackBy: tackByFn"
+          class="list__item"
+          [ngClass]="{ 'list__item--active': isSelected(type.id!!) }"
+        >
+          <button (click)="handleCategory(type, i + 1)">
+            {{ type.id }}: {{ type.name }} <span [ngClass]="{ 'list__item--more': hasNext(type) }"></span>
```
Comment (andrewf-bjss) [inline src/app/forms/components/test-type-select/test-type-select.component.html:47]:
```suggestion
            {{ type.name }} <span [ngClass]="{ 'list__item--more': hasNext(type) }"></span>
```

**Review by andrewf-bjss — APPROVED** [review]

**Review by mcadenas-bjss — DISMISSED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Review by Ruairidh-BJSS — APPROVED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Review by andrewf-bjss — APPROVED** [review]

**Review by Ruairidh-BJSS — APPROVED** [review]

**Pull request merged**